cli/internal/context/context.go

```diff
@@ -419,3 +419,30 @@ func (c *Context) resolveDepGraph(wg *errgroup.Group, workspace *fs.PackageJSON,
 		})
 	}
 }
+
+// TopologicalGraphAncestors returns the ancestors of a set of nodes in the workspace graph.
```

Comment:
I think it's fine to have this include the source nodes in the returned set, but since this is used for a special case we should change the name to `InternalDependencies`.